### PR TITLE
Issues retrieval from PR body

### LINKED ISSUE #499: Working only for prs to default branch
If you create the pr with main/master (default branch) as the base the closingIssuesReferences does return the issue.
As the issue on GitHub is only closed if it is merged into the default branch, this makes sense.

Is there a way to make this workflow work also for prs where the base branch is not the default branch? (Maybe via timelineItems or pr body?)
Inspiration:
https://github.com/marketplace/actions/verify-linked-issue

### PR DESCRIPTION
**Pull request by synapse**
Closes #499 
Added a new utility that will parse the PR body content and search for related issue. Just as per the [GH docs](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) the script checks for the same words to close the issue. e.g.: `close #123`

### CODE REVIEW

**Review by simoneb — CHANGES_REQUESTED** [review]
This needs to be disabled by default, to adhere to GitHub's default behavior, and can be opted-in with a new configuration option. In other words, it should not replace the default behavior unless this option is enabled.

There are also some nuances to this, let's take it offline.

**Comment by simoneb** [comment]
As discussed, let's introduce a new action input called `loose-matching` which will handle:

- as you already did in this PR, the built-in GitHub keywords (such as Closes, Fixes, ...) followed by an issue number formatted as "#123". This will search for an issue in the same repo and, if found, will consider the issue linked from the PR
- a new case that you're currently not handling, which supports linking an issue using the same keywords, followed by an url. In this case, we will parse the url, retrieve the owner, the repo and the issue ID, and then we'll consider the issue linked if we successfully find the issue, using the same github API we are using in the first case

Note that these behaviors should be opt-in, meaning that they're carried out only if the `loose-matching` input is true.

Let's also update the docs to include the new input in the list, and maybe a short section to explain in more detail (but brefly) the enabling the input does.

File: src/action.js [hunk 1]
```diff
@@ -55,10 +56,27 @@ async function run() {
...
+        repoOwner: owner.login,
+        octokit,
+      });
+      linkedIssuesCount = issues.length;
+    } else {
+      linkedIssuesCount = pullRequest?.closingIssuesReferences?.totalCount;
+      issues = (pullRequest?.closingIssuesReferences?.nodes || []).map(
+        (node) => `${node.repository.nameWithOwner}#${node.number}`
+      );
+    }
```
Comment (simoneb) [inline src/action.js:79]: let's extract this in a function to avoid cluttering the main logic with low level details

File: README.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ GitHub action to check if pull requests have their corresponding issues linked,
...
 | `comment`                  | No | `true`              | A boolean value that allow the action to create a comment. |
 | `custom-body-comment`      | No | "No linked issues found. Please add the corresponding issues in the pull request description. <br/> [Use GitHub automation to close the issue when a PR is merged](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)" | Custom body PR comment. |
+| `loose-matching`                  | No | `false`              | A boolean value indicating whether the action should verify linked issues on branches other than the default one. This option supports both local issues (e.g., `#123`) and external ones (e.g., `https://github.com/org-name/repo/issues/123`). You can associate a pull request with an issue by including a supported keyword in the pull request's description (e.g., `fixes #123`). Check the GitHub documentation for a list of [supported keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).|
```
Comment (simoneb) [inline README.md:14]: I would reword this slightly differently, because the description suggests that the primary purpose of this option is to support PRs which merge on non-default branches, but the purpose is two fold: that and also supporting external issues, regardless of whether they merge into the default branch or not.

File: src/action.js (hunk 1 above)
Comment (synapse) [reply src/action.js]: done

**Review by simoneb — APPROVED** [review]
